lib/dtutils/file.lua

```diff
@@ -0,0 +1,523 @@
+--[[
+
+    dtutils/file.lua - common darktable lua file functions
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+    Copyright (C) 2016 Tobias Jakobs
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local dtutils_file = {}
+dtutils_file.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "License"},
+  Name = [[dtutils.file - common darktable lua file functions]],
+  Synopsis = [[local df = require "lib/dtutils.file"]],
+  Description = [[dtutils.file provides common file manipulation functions used in
+  constructing Darktable lua scripts]],
+  License = [[This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.]],
+  functions = {}
+}
+
+local dt = require "darktable"
+
+local log = require "lib/libLog"
+
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("dtutils.file",dt.configuration.config_dir.."/lua/locale/")
```

Comment:
Maybe I don't understand gettext correctly.  My understanding of this line is that it tells where the .mo files, that are generated from the .po files in the script directories, are, not the path to the .po files.  Does darktable read the .po files and use those directly?  I wrote another tool, gen_mo.lua, to find all of the .po files, compile them to .mo files and place those in dt.configuration.config_dir .. "/lua/locale/", which I create.  Dumping ... just gave me the relative path, i.e. from lib/dtutils.lua I got lib/dtuils.  I wrote a dtutils.pwd() function to get the current directory but then scrapped it when I thought I had gettext figured out.  I'll look at the darktable code and see what it's doing...